staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -878,6 +889,15 @@ func (r *Request) request(fn func(*http.Request, *http.Response)) error {
 	}
 }
 
+func (r *Request) injectTrace() opentracing.Span {
+	if r.ctx == nil {
+		r.ctx = context.Background()
+	}
+	span, ctx := opentracing.StartSpanFromContext(r.ctx, fmt.Sprintf("%v %v", r.verb, r.resource))
```

Comment:
I had a look, the existing code there is too coupled. I think it's simpler to add the tags explicitly, rather than pull another dependency just for that. You want at least these:

```go
	ext.HTTPMethod.Set(span, req.Method)
	ext.HTTPUrl.Set(span, req.URL.String())
	ext.SpanKindRPCClient.Set(span)
	ext.Component.Set(span, componentName) // optional
```

If you have the peer/callee address easily accessible, it's also useful to set

```go
        ext.PeerAddress.Set(span, address) // IP string or anything, really
```